workspace_host/interface.sql

```diff
@@ -37,3 +37,68 @@ FROM
 WHERE
     w.id = $workspace_id
     AND w.workspace_host_id = wh.id;
+
+-- BLOCK get_workspace
+SELECT
+    w.*
+FROM
+    workspaces AS w
+WHERE
+    w.id = $workspace_id;
```

Comment:
For safety, we could also pass in `$instance_id` here and make sure that matches. Of course this shouldn't be needed, but if PL ever gets confused about which host gets which message (maybe a race condition?) then that would help contain the damage.

Reply:
Actually, we may as well change all the queries here (including the UPDATEs) to restrict to `$instance_id`